kornia/geometry/camera/stereo.py

```diff
@@ -0,0 +1,287 @@
+    Args:
+        disparity_tensor (torch.Tensor): The disparity tensor of shape :math:`(B, H, W)`.
+    """
+    if disparity_tensor.ndim != 3:
+        raise StereoException(f"Expected 'disparity_tensor' to have 3 dimensions." f"Got {disparity_tensor.ndim}.")
+
+    if disparity_tensor.dtype != torch.float32:
+        raise StereoException(
+            f"Expected 'disparity_tensor' to have dtype torch.float32." f"Got {disparity_tensor.dtype}"
+        )
+
+
+def _check_Q_matrix(Q_matrix: torch.Tensor):
+    """
+    Utility function to ensure Q matrix is of correct form.
+    Args:
+        Q_matrix (torch.Tensor): The Q matrix for reprojecting disparity to a point cloud of shape :math:`(B, 4, 4)`
+    """
+    if not Q_matrix.ndim == 3:
+        raise StereoException(f"Expected 'Q_matrix to have 3 dimenstions." f"Got {Q_matrix.ndim}")
+
+    if not Q_matrix.shape[1:] == (4, 4):
+        raise StereoException(
+            f"Expected last two dimensions of 'Q_matrix' to be of shape (4, 4)." f"Got {Q_matrix.shape}"
+        )
+
+    if not Q_matrix.dtype == torch.float32:
+        raise StereoException(f"Expected 'Q_matrix' to be of type torch.float32." f"Got {Q_matrix.dtype}")
+
+
+def reproject_disparity_to_3D(disparity_tensor: torch.Tensor, Q_matrix: torch.Tensor) -> torch.Tensor:
+    """
+    Reproject the disparity tensor to a 3D point cloud.
+
+    Args:
+        disparity_tensor (torch.Tensor): Disparity tensor of shape :math:`(B, H, W)`.
+        Q_matrix (torch.Tensor): Tensor of Q matrices of shapes :math:`(B, 4, 4)`.
+
+    Returns:
+        torch.Tensor: The 3D point cloud of shape :math:`(B, H * W, 3)`
+    """
+    _check_Q_matrix(Q_matrix)
+    _check_disparity_tensor(disparity_tensor)
+
+    batch_size, rows, cols = disparity_tensor.shape
+    dtype = disparity_tensor.dtype
+    device = disparity_tensor.device
+    homogenous_observation_ndim = 4
+    euclidian_observation_ndim = homogenous_observation_ndim - 1
+
+    # Construct a mesh grid of uv values, i.e. the tensors will contain 1:rows and 1:cols.
+    u, v = torch.meshgrid(
+        torch.arange(rows, dtype=dtype, device=device), torch.arange(cols, dtype=dtype, device=device)
+    )
+    u, v = u.expand(batch_size, -1, -1), v.expand(batch_size, -1, -1)
+
+    # The z dimension in homogenous coordinates are just 1.
+    z = torch.ones((batch_size, rows, cols), dtype=dtype, device=device)
+
+    # Stack the observations into a tensor of shape (batch_size, 4, -1) that contains all
+    # 4 dimensional vectors [u v disparity 1].
+    uvdz = torch.stack((u, v, disparity_tensor, z), 1).reshape(batch_size, homogenous_observation_ndim, -1)
+
+    # Matrix multiply all vectors with the Q matrix
+    hom_points = torch.bmm(Q_matrix, uvdz)
+
+    # Convert from homogenous to euclidian space.
+    z_points = torch.unsqueeze(hom_points[:, euclidian_observation_ndim], 1)
+    points = (hom_points / z_points)[:, :euclidian_observation_ndim]
```

Comment:
we have `kornia.geometry.convert_points_from_homogeneous` that takes care of zero division